class/DatabaseConnection.php

```diff
@@ -0,0 +1,340 @@
+require_once('error_helper.php');
+
+/**
+* Super Class
+*
+* @author Smruti Ranjan
+*/
+class DatabaseConnection
+{
+	public static $conn = NULL;
+	public static $connection_obj = NULL;
+
+	/**
+	* Constructor to create a database connection
+	*
+	* @access public
+	* @param  void
+	* @return void[]
+	*/
+	private function __construct($db_param)
+	{
+		self::$conn = mysqli_connect($db_param['hostname'], $db_param['username'], 
+			$db_param['password'], $db_param['database']);
+
+		if(mysqli_connect_errno(self::$conn))
+		{
+			log_error('Creation of databse connection');
+			die ('Failed to connect to Database, Please try later :' . mysqli_connect_error());
+		}
+	}
+
+	/**
+	* To check of exixting connection object
+	*
+	* @access public
+	* @param  void
+	* @return object
+	*/
+	public static function create_connection($db_param)
+	{
+		if(self::$connection_obj == NULL)
+		{
+			self::$connection_obj = new DatabaseConnection($db_param);
+		}
+
+		return self::$connection_obj;
+	}
+
+	/**
+	* To execute SQL queries
+	*
+	* @access public
+	* @param  string $query
+	* @return object
+	*/
+	public static function db_query($query)
+	{
+		$result = mysqli_query(self::$conn, $query);
+		return $result;
+	}
+
+	/**
+	* To fetch executed select query
+	*
+	* @access public
+	* @param  object $passed_object
+	* @return object
+	*/
+	public static function db_fetch_array($passed_object)
+	{
+		$result = mysqli_fetch_array($passed_object, MYSQLI_ASSOC);
+		return $result;
+	}
+
+	/**
+	* To fetch number of rows in a db object
+	*
+	* @access public
+	* @param  object $object
+	* @return integer
+	*/
+	public static function db_num_rows($object)
+	{
+		$rows = mysqli_num_rows($object);
+		return $rows;
+	}
+
+	/**
+	* To select all data from the database
+	*
+	* @access public
+	* @param  integer $id
+	* @return object
+	*/
+	public function select($id)
+	{
+		$q_fetch = "SELECT emp.first_name AS f_name, emp.middle_name AS m_name, 
+			emp.last_name AS l_name, emp.email AS email, emp.prefix AS prefix, emp.gender AS gender, 
+			emp.dob AS dob, emp.marital_status AS marital_status, emp.employment AS employment, 
+			emp.employer AS employer, res.street AS r_street, res.city AS r_city, 
+			res.state AS r_state, res.zip AS r_zip, res.phone AS r_phone, 
+			res.fax AS r_fax, off.street AS o_street, off.city AS o_city, off.state AS o_state, 
+			off.zip AS o_zip, off.phone AS o_phone, off.fax AS o_fax, emp.photo AS photo, 
+			emp.extra_note AS notes, emp.comm_id AS comm_id 
+			FROM employee AS emp 
+			INNER JOIN address AS res ON (emp.id = res.emp_id AND res.address_type = 'residence')
+			INNER JOIN address AS off ON (emp.id = off.emp_id AND off.address_type = 'office')
+			WHERE emp.id = $id";
+
+		$result_select = DatabaseConnection::db_query($q_fetch);
+		$row = DatabaseConnection::db_fetch_array($result_select);
+		return $row;
+	}
+
+	/**
+	* To select the desired eamil from the database
+	*
+	* @access public
+	* @param  integer $id
```

Comment:
Function has two parameters, specify both in function header
@param string   $email
@param integer $id